### PR TITLE
temporary melodic

### PR DESCRIPTION
**Pull request by carebare47**
## Proposed changes

Please explain the changes you made. Add pictures or videos to explain them better if appropriate.

## Checklist

If the task is applicable to this pull request (see applicability criteria in brackets), make sure it is completed before checking the corresponding box. Otherwise, tick the box right away. Make sure that **ALL** boxes are checked **BEFORE** the PR is merged.

- [x] Manually tested that added code works as intended (if any functional/runnable code is added).
- [x] Added automated tests (if a new class is added (Python or C++), interface of that class must be unit tested).
- [x] Tested on real hardware (if the changed or added code is used with the real hardware).
- [x] Added documentation (For any new feature, explain what it does and how to use it. Write the documentation in a relevant space, e.g. Github, Confluence, etc.)

### CODE REVIEW

File: sr_test_force_resolution/package.xml
Comment (giusebar) [inline sr_test_force_resolution/package.xml]: v 0.0.1

File: sr_test_force_resolution/package.xml
Comment (giusebar) [inline sr_test_force_resolution/package.xml]: clean up this a bit

File: sr_test_force_resolution/src/sr_test_force_resolution/sr_test_force_resolution.py
Comment (giusebar) [inline sr_test_force_resolution/src/sr_test_force_resolution/sr_test_force_resolution.py]: we should get the params locally. and launch this via a launch file or at least be able to pass the arguments via terminal.

Comment (giusebar) [reply sr_test_force_resolution/src/sr_test_force_resolution/sr_test_force_resolution.py]: e.g. side = left can only be set by modifying it here.

File: sr_test_force_resolution/src/sr_test_force_resolution/sr_test_force_resolution.py
Comment (giusebar) [inline sr_test_force_resolution/src/sr_test_force_resolution/sr_test_force_resolution.py]: can we get this from some parameters? it would be more flexible

File: sr_test_force_resolution/src/sr_test_force_resolution/sr_test_force_resolution.py
Comment (giusebar) [inline sr_test_force_resolution/src/sr_test_force_resolution/sr_test_force_resolution.py]: is this max min because each finger has weird different signs?

File: sr_test_force_resolution/src/sr_test_force_resolution/sr_test_force_resolution.py
Comment (giusebar) [inline sr_test_force_resolution/src/sr_test_force_resolution/sr_test_force_resolution.py]: we should move these const outside

File: sr_test_force_resolution/src/sr_test_force_resolution/sr_test_force_resolution.py
Comment (giusebar) [inline sr_test_force_resolution/src/sr_test_force_resolution/sr_test_force_resolution.py]: We should split this run function in multiple functions. it would be good to have e.g.  some of the if instead of like that passed through a funciton. and some of the for loops can also be wrap in functions

File: sr_test_force_resolution/src/sr_test_force_resolution/sr_test_force_resolution.py
Comment (giusebar) [inline sr_test_force_resolution/src/sr_test_force_resolution/sr_test_force_resolution.py]: e.g. having a funciton that takes as params (joint, input_angle, mode)

File: sr_test_force_resolution/src/sr_test_force_resolution/sr_test_force_resolution.py
Comment (giusebar) [inline sr_test_force_resolution/src/sr_test_force_resolution/sr_test_force_resolution.py]: why First finger is different here?

File: sr_test_force_resolution/src/sr_test_force_resolution/sr_test_force_resolution.py
Comment (giusebar) [inline sr_test_force_resolution/src/sr_test_force_resolution/sr_test_force_resolution.py]: wrap this line in some nice function with name that clearly states what it does

File: sr_test_force_resolution/src/sr_test_force_resolution/sr_test_force_resolution.py
Comment (giusebar) [inline sr_test_force_resolution/src/sr_test_force_resolution/sr_test_force_resolution.py]: lost track of which else is this :D

File: sr_test_force_resolution/src/sr_test_force_resolution/sr_test_force_resolution.py
Comment (giusebar) [inline sr_test_force_resolution/src/sr_test_force_resolution/sr_test_force_resolution.py]: improve name, and also here would be good to wrap the long line in a function with clear name

File: sr_test_force_resolution/src/sr_test_force_resolution/sr_test_force_resolution.py
Comment (giusebar) [inline sr_test_force_resolution/src/sr_test_force_resolution/sr_test_force_resolution.py]: imrpve name, not sure what l or lst is

File: sr_test_force_resolution/src/sr_test_force_resolution/sr_test_force_resolution.py
Comment (giusebar) [inline sr_test_force_resolution/src/sr_test_force_resolution/sr_test_force_resolution.py]: why do we need file_prefix? is this the description of the test?

**Review by giusebar — CHANGES_REQUESTED** [review]

File: sr_test_force_resolution/src/sr_test_force_resolution/sr_test_force_resolution.py
Comment (carebare47) [reply sr_test_force_resolution/src/sr_test_force_resolution/sr_test_force_resolution.py]: We could generate this dict automatically. The thinking here was that someone could easily change the joint limits for various tests

File: sr_test_force_resolution/src/sr_test_force_resolution/sr_test_force_resolution.py
Comment (carebare47) [reply sr_test_force_resolution/src/sr_test_force_resolution/sr_test_force_resolution.py]: kinda, it's so that if we want to test a J4, this matrix tells the other fingers how to get out of the way so that j4 can be tested through it's full range

File: sr_test_force_resolution/src/sr_test_force_resolution/sr_test_force_resolution.py
Comment (carebare47) [reply sr_test_force_resolution/src/sr_test_force_resolution/sr_test_force_resolution.py]: yeah, it's an optional description you can add so that finding your data later is easier

File: sr_test_force_resolution/src/sr_test_force_resolution/sr_test_force_resolution.py
Comment (mikramarc) [inline sr_test_force_resolution/src/sr_test_force_resolution/sr_test_force_resolution.py:33]: private attributes with leading underscore

File: sr_test_force_resolution/nodes/sr_test_force_resolution
Comment (mikramarc) [inline sr_test_force_resolution/nodes/sr_test_force_resolution]: please add __name__=__main__ stuff

File: sr_test_force_resolution/nodes/sr_test_force_resolution
Comment (mikramarc) [inline sr_test_force_resolution/nodes/sr_test_force_resolution]: please throw exception, exit(0) means exit with no issues

File: sr_test_force_resolution/src/sr_test_force_resolution/sr_test_force_resolution.py
Comment (mikramarc) [inline sr_test_force_resolution/src/sr_test_force_resolution/sr_test_force_resolution.py]: anything that is const, please make CONST_VARIABLE convention

File: sr_test_force_resolution/src/sr_test_force_resolution/sr_test_force_resolution.py
Comment (mikramarc) [inline sr_test_force_resolution/src/sr_test_force_resolution/sr_test_force_resolution.py:44]: if its just one line, I'd say not much use for a separte method, but i'll leave it up to your taste

File: sr_test_force_resolution/src/sr_test_force_resolution/sr_test_force_resolution.py
Comment (mikramarc) [inline sr_test_force_resolution/src/sr_test_force_resolution/sr_test_force_resolution.py]: underscores. please check the whole pr

File: sr_test_force_resolution/src/sr_test_force_resolution/sr_test_force_resolution.py
Comment (mikramarc) [inline sr_test_force_resolution/src/sr_test_force_resolution/sr_test_force_resolution.py]: self.cmd is a char not an array, shouldnt be == instead of in?

File: sr_test_force_resolution/src/sr_test_force_resolution/sr_test_force_resolution.py
Comment (mikramarc) [inline sr_test_force_resolution/src/sr_test_force_resolution/sr_test_force_resolution.py]: you're alraedy doing this in validate method

File: sr_test_force_resolution/src/sr_test_force_resolution/sr_test_force_resolution.py
Comment (mikramarc) [inline sr_test_force_resolution/src/sr_test_force_resolution/sr_test_force_resolution.py:120]: please add some empty lines for visual separation of different operations

File: sr_test_force_resolution/src/sr_test_force_resolution/sr_test_force_resolution.py
Comment (mikramarc) [inline sr_test_force_resolution/src/sr_test_force_resolution/sr_test_force_resolution.py]: please put that in a yaml file instead of keeping in code.

Comment (mikramarc) [reply sr_test_force_resolution/src/sr_test_force_resolution/sr_test_force_resolution.py]: same for joint states zero

File: sr_test_force_resolution/src/sr_test_force_resolution/sr_test_force_resolution.py
Comment (mikramarc) [inline sr_test_force_resolution/src/sr_test_force_resolution/sr_test_force_resolution.py]: yaml again

File: sr_test_force_resolution/src/sr_test_force_resolution/sr_test_force_resolution.py
Comment (mikramarc) [inline sr_test_force_resolution/src/sr_test_force_resolution/sr_test_force_resolution.py]: lets make run a public method and move it our of init. taht way user can decide when importring the class to run it in a loop or not

File: sr_test_force_resolution/src/sr_test_force_resolution/sr_test_force_resolution.py
Comment (mikramarc) [inline sr_test_force_resolution/src/sr_test_force_resolution/sr_test_force_resolution.py]: no need for two log infos ina  row

File: sr_test_force_resolution/src/sr_test_force_resolution/sr_test_force_resolution.py
Comment (mikramarc) [inline sr_test_force_resolution/src/sr_test_force_resolution/sr_test_force_resolution.py:270]: this is literally the same code as about apart from the positon and effort words. Please use shared method with parameter

File: sr_test_force_resolution/src/sr_test_force_resolution/sr_test_force_resolution.py
Comment (mikramarc) [inline sr_test_force_resolution/src/sr_test_force_resolution/sr_test_force_resolution.py]: same thing

File: sr_test_force_resolution/src/sr_test_force_resolution/sr_test_force_resolution.py
Comment (mikramarc) [inline sr_test_force_resolution/src/sr_test_force_resolution/sr_test_force_resolution.py]: debug code?

File: sr_test_force_resolution/src/sr_test_force_resolution/sr_test_force_resolution.py
Comment (mikramarc) [inline sr_test_force_resolution/src/sr_test_force_resolution/sr_test_force_resolution.py]: ieh??

File: sr_test_force_resolution/src/sr_test_force_resolution/sr_test_force_resolution.py
Comment (mikramarc) [inline sr_test_force_resolution/src/sr_test_force_resolution/sr_test_force_resolution.py:310]: what does free_j4 mean?

File: sr_test_force_resolution/src/sr_test_force_resolution/sr_test_force_resolution.py
Comment (mikramarc) [inline sr_test_force_resolution/src/sr_test_force_resolution/sr_test_force_resolution.py]: remove all of those please

**Review by mikramarc — CHANGES_REQUESTED** [review]

File: sr_test_force_resolution/src/sr_test_force_resolution/sr_test_force_resolution.py
Comment (carebare47) [reply sr_test_force_resolution/src/sr_test_force_resolution/sr_test_force_resolution.py]: it's to try and clear up the console output a bit

File: sr_test_force_resolution/src/sr_test_force_resolution/sr_test_force_resolution.py
Comment (carebare47) [reply sr_test_force_resolution/src/sr_test_force_resolution/sr_test_force_resolution.py:310]: Say we want to test MFJ4 through it's full range, first we've gotta move FFJ4, RFJ4 and FFJ4 out of the way

**Comment by carebare47** [comment]
This needs some more work done before merging:
 * simplify it  (e.g. joint zeros and min/max should be gotten from the param server instead of being stored in those yaml files)
 * interface isn't very good
 * code needs testing with up-to-date noetic

Closing this PR for now

**Pull request closed without merging**
